### PR TITLE
fixes #14449 - run facter command to check FQDN, support AIO path

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

**Comment by ekohl** [comment]
:+1:

File: checks/hostname.rb [hunk 1]
```diff
@@ -32,8 +26,11 @@ def error_exit(message, code)
   exit code
 end
 
+ENV['PATH'] = ENV['PATH'].split(File::PATH_SEPARATOR).concat(['/opt/puppetlabs/bin']).join(File::PATH_SEPARATOR)
+facter_fqdn = `facter fqdn`
+
 # Check that facter actually has a value that matches the hostname.
 # This should always be true for facter >= 1.7
-error_exit(DIFFERENT + BASE, 1) if Facter.value(:fqdn) != `hostname -f`.chomp
+error_exit(DIFFERENT + BASE, 1) if facter_fqdn != `hostname -f`.chomp
```
Comment (mmoll) [inline checks/hostname.rb:34]: I think a chomp is also needed for facter_fqdn:

```
irb(main):001:0> `facter --debug fqdn`
=> "emperor.kvedulv.de\n"
irb(main):002:0> 
```

Comment (domcleal) [inline checks/hostname.rb:34]: Ah yes, thanks, I reproduced the issue.  I've added a chomp to the line where facter_fqdn's set.

**Comment by mmoll** [comment]
merged, thanks @domcleal!

**Pull request merged**